system/include/emscripten/promise.h

```diff
@@ -135,6 +135,19 @@ __attribute__((warn_unused_result)) em_promise_t emscripten_promise_any(
 __attribute__((warn_unused_result)) em_promise_t
 emscripten_promise_race(em_promise_t* promises, size_t num_promises);
 
+// Suspend the current Wasm execution context until the given promise has been
+// settled.
+//
+// Since the stack is not unwound while Wasm execution is suspended, it is
+// safe to pass pointers to the stack to asynchronous work that is waited on
+// with this function.  If an exception occurs while the promise handler is
+// running then the suspended Wasm will never be resumed and this function
+// will never return.
```

Comment:
Looks like this part of the comment needs to be updated for this new approach.

Reply:
Do exceptions automatically turn into promise rejections?